### PR TITLE
doc: Update release notes for issues closed in this release

### PR DESCRIPTION
**Pull request by galak**
This list was created by what issues where closed from Sept 7th and everything tagged with either a 1.10 milestone or no milestone at all.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by galak** [comment]
Also this is just the issues closed, not the PRs.

**Review by nashif — APPROVED** [review]

File: doc/release-notes-1.10.rst [hunk 1]
```diff
@@ -171,5 +171,353 @@ Issue Related Items
 
 .. comment  List derived from Jira/GitHub Issue query: ...
 
-* :jira:`ZEP-248` - reference a (public) Jira issue
-* :github:`1234` - reference a GitHub issue
+* :github:`1280` - shell on Arduino Due prints "shell>" before the delayed boot banner
```
Comment (dbkinder) [inline doc/release-notes-1.10.rst]: How about sorting by issue number?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/release-notes-1.10.rst (hunk 1 above)
Comment (galak) [reply doc/release-notes-1.10.rst]: I can, they are currently sorted by closed date (not that anyone knows that).

Comment (dbkinder) [reply doc/release-notes-1.10.rst]: I was also surprised that Sphinx didn't complain about underscores.  Things like ``spi_legacy.h`` look like a hyperlink reference to something named ``spi``, so we've had to escape that by writing ``spi\_legacy.h``.  But didn't see the CI build failing, so guess the current version of Sphinx is more clever than previous versions.

**Review by dbkinder — APPROVED** [review]
Much nicer, thanks.

**Pull request merged**